feat(data/fintype): prove `card (subtype p) ≤ card α`

urkud opened the pull request:
TO CONTRIBUTORS:

Make sure you have:

  * [X] reviewed and applied the coding style: [coding](https://github.com/leanprover/mathlib/blob/master/docs/contribute/style.md), [naming](https://github.com/leanprover/mathlib/blob/master/docs/contribute/naming.md)
  * [X] reviewed and applied [the documentation requirements](https://github.com/leanprover/mathlib/blob/master/docs/contribute/doc.md)
  * [ ] for tactics:
     * [ ] added or adapted documentation in [tactics.md](https://github.com/leanprover/mathlib/blob/master/docs/tactics.md)
     * [ ] write an example of use of the new feature in [tactics.lean](https://github.com/leanprover/mathlib/blob/master/test/tactics.lean)
  * [X] make sure definitions and lemmas are put in the right files
  * [X] make sure definitions and lemmas are not redundant

If this PR is related to a discussion on Zulip, please include a link in the discussion.

For reviewers: [code review check list](https://github.com/leanprover/mathlib/blob/master/docs/contribute/code-review.md)

urkud committed:
feat(data/fintype): prove `card (subtype p) ≤ card α`

jcommelin reviewed (commented):
The cardinal of a subtype will always be smaller than the cardinal of the ambient type. I guess would make sense to also add that lemma.

urkud committed:
Add `cardinal.mk_le_of_subtype`

urkud committed:
Merge branch 'master' into card-subtype-le

urkud committed:
Merge branch 'master' into card-subtype-le

rwbarton commented on src/set_theory/cardinal.lean:
```diff
@@ -74,6 +74,9 @@ theorem mk_le_of_injective {α β : Type u} {f : α → β} (hf : injective f) :
 theorem mk_le_of_surjective {α β : Type u} {f : α → β} (hf : surjective f) : mk β ≤ mk α :=
 ⟨embedding.of_surjective hf⟩
 
+theorem mk_le_of_subtype {α : Type u} (p : α → Prop) : mk (subtype p) ≤ mk α :=
```
This already basically exists as `mk_subtype_le`, but maybe it doesn't hurt to have a version with `subtype` instead of a coercion?

urkud replied:
How did I miss `mk_subtype_le`? I'd remove `mk_le_of_subtype` (why did I choose such a bad name?).

urkud replied:
We had both `mk_subtype_le` and `mk_set_le` with the same signature. Now `mk_set_le` takes a set, and `mk_subtype_le` takes a map `α → Prop`.

urkud committed:
Rename `mk_le_of_subtype` to `mk_subtype_le`, use it in `mk_set_le`

Earlier both `mk_subtype_le` and `mk_set_le` took `set α` as an
argument. Now `mk_subtype_le` takes `α → Prop`.

urkud committed:
Merge branch 'master' into card-subtype-le
